client-core/src/wallet/syncer.rs

```diff
@@ -106,6 +106,12 @@ pub struct SyncerConfig<S: SecureStorage, C: Client> {
     block_height_ensure: u64,
 }
 
+#[derive(Clone)]
+pub struct SyncCallback {
+    pub user_data: u64,
+    pub user_callback: extern "C" fn(u64, u64, u64, u64) -> i32,
+}
```

Comment:
You should use a rust closure here, wrap up the c ffi stuff in the c lib.

Reply:
i tried, some strange issues, 
so i changed back to function again.

Reply:
@leejw51crypto do you still have the code? you can possibly share on slack or via vscode live

Reply:
i talked with yi,
modifying now
1. adding closure
2. use void* instead of u64
not to use ffi code in core